modules/store/src/models.ts

```diff
@@ -75,17 +75,11 @@ export type FunctionIsNotAllowed<
 export type Creator<
   P extends any[] = any[],
   R extends object = object
-> = R extends any[]
-  ? ArraysAreNotAllowed
-  : R extends { type: any }
-    ? TypePropertyIsNotAllowed
-    : FunctionWithParametersType<P, R>;
+> = FunctionWithParametersType<P, R>;
 
-export type PropsReturnType<T extends object> = T extends any[]
+export type NotAllowedCheck<T extends object> = T extends any[]
```

Comment:
What is the difference between this type and `DisallowArraysAndTypeProperty` ? 
I believe `DisallowArraysAndTypeProperty` isn't used with this refactor, so we can also remove that type or mark it to be deleted in v9

Reply:
Did realize that we got `DisallowArraysAndTypeProperty` still hanging around. I removed it in favor of `NotAllowedCheck`, as it wasn't exported to the public api.